File: 백준/Silver/2164. 카드2/카드2.cc
#include <iostream>
#include <queue>

using namespace std;

int main() {
    int n;
    cin >> n;

    queue<int> que;
    for (int i = 1; i <= n; i++)
        que.push(i);

    while (que.size() > 1) {
        que.pop();
        que.push(que.front()); que.pop();
    }

    cout << que.front();
}

File: 백준/Silver/2164. 카드2/카드2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.LinkedList;
import java.util.Queue;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        if (n == 1) {
            System.out.println(1);
            return;
        }

        Queue<Integer> que = new LinkedList<>();
        for (int i = 1; i <= n; i++) {
            que.offer(i);
        }

        while (!que.isEmpty()) {
            que.poll();
            if (que.size() == 1) {
                break;
            }

            que.offer(que.poll());
        }

        System.out.print(que.poll());
    }
}